### PR TITLE
Min event threshold

### PR DESCRIPTION
**Pull request by rslish**
Added model parameter to set minimum length of VAD event before the event is triggered.

### CODE REVIEW

File: cpp/detector/vad_model.h [hunk 1]
```diff
@@ -32,12 +32,17 @@ namespace cobalt
...
+        {
+            return mMinEventThreshold;
+        }
 
     private:
         // members
         ConfigSet::Ptr mConfigset;
         kaldi::VadEnergyOptions mVadConfig;
         std::string mVadType;
+        int mMinEventThreshold;
```
Comment (shkyang) [inline cpp/detector/vad_model.h]: could be const.

File: cpp/util/vad_events.cpp [hunk 2]
```diff
@@ -44,35 +44,53 @@ namespace cobalt
...
+                    else
+                    {
+                        event.vadEventType = EOS;
+                    }
+                    stagedEvents.push_back(event);
                 }
-                events.push_back(event);
                 previousState = currentState;
             }
+            else if (stagedEvents.size() > 0)
```
Comment (shkyang) [inline cpp/util/vad_events.cpp]: can we just move this check outside the loop?

File: cpp/util/test_util.cpp [hunk 3]
```diff
@@ -110,26 +110,31 @@ void testVadEventsFromFrames()
         frameVadInfo.push_back(true);
     }
     VadEvents events;
+    VadEvents stagedEvents;
```
Comment (shkyang) [inline cpp/util/test_util.cpp]: can we add some new tests that'll illustrate different use-cases for staged events?

And any edge cases you can think of?

**Comment by shkyang** [comment]
requests for more comments and tests.

File: cpp/util/test_util.cpp [hunk 4]
```diff
@@ -140,12 +140,126 @@ void testVadEventsFromFrames()
...
+    }
+    VadEvents events;
+    VadEvents stagedEvents;
+    int minEventThresh = 100;
+    int previousEndFrame = 0;
+    bool previousState = false;
+    // 1 Frame short of thresh
+    frameVadResultsToEvents(frameVadInfo, events, stagedEvents, minEventThresh, previousEndFrame, previousState);
+    cout << endl << "events with previousEndFrame " << previousEndFrame << " previousState " << previousState << endl;
+    printVadEvents(events);
```
Comment (shkyang) [inline cpp/util/test_util.cpp]: I think we can use some asserts here, like assert on # of events you got back.

File: cpp/util/vad_events.cpp [hunk 5]
```diff
@@ -49,19 +49,24 @@ namespace cobalt
...
+            // Change in state means an event
             if (currentState != previousState)
             {
-                // If last event is in stagedEvents, it was not longer than threshold
+                // Events are removed from stagedEvents once they reach the threshold
+                // So if last event is still in stagedEvents, it was not longer than threshold
+                // Clear stagedEvent; no new event staged--we're in same state as last event in events
                 if (stagedEvents.size() > 0)
                 {
                     stagedEvents.clear();
```
Comment (shkyang) [inline cpp/util/vad_events.cpp]: so if stagedEvents is always size 1 or 0, can we just make it previousEvent? You'd need to initialize it to some state for the first time around.

File: cpp/util/vad_events.cpp [hunk 6]
```diff
@@ -44,7 +44,7 @@ namespace cobalt
         throw std::runtime_error("kaldi vad was neither 0 or 1.");
     }
 
-    void frameVadResultsToEvents(const std::vector<bool> &vadResults, VadEvents& events, VadEvents& stagedEvents, int minEventThresh, int previousEndFrame, bool previousState)
+    void frameVadResultsToEvents(const std::vector<bool> &vadResults, VadEvents& events, VadEvent& previousEvent, bool& prevEventAdded, int minEventThresh, int previousEndFrame, bool previousState)
```
Comment (shkyang) [inline cpp/util/vad_events.cpp]: no need for reference on bool.

**Comment by shkyang** [comment]
🚢 
looks good, change the reference to pass by value on bool.
